### ISSUE TITLE
Support puppeteer 5

### ISSUE THREAD

**Issue by Niek** [berstend/puppeteer-extra#228]
Puppeteer 4  (*edit*: 5 now) has been released. The changes are quite trivial for puppeteer-extra, I think the most important is to remove this:
https://github.com/berstend/puppeteer-extra/blob/a2a06f971973067a631b1156cf826a034b1323ee/packages/puppeteer-extra-plugin-stealth/index.js#L155

I can do a PR but I'm not sure if this repo is maintained anymore @berstend?

**Comment by brunogaspar** [comment]
Author is MIA since end of February, so i guess sending a pull request is just going to sit there, unfortunately.

I've a private "fork" of this with some modifications that helped me in some scenarios, but not sure if there's much interest in making it public or even for the community to help out in maintaining/moving forward.

**Comment by momala454** [comment]
i thnk people are interested in your mods

**Comment by brunogaspar** [comment]
Alright, i'll look into making it public and how to make it installable through NPM, not much experience with it, just Composer for PHP :)

**Comment by Niek** [comment]
For those who're interested: I decided to make a similar package based on Playwright instead: https://github.com/Niek/playwright-addons. It's still in development, at the moment it includes an ad blocker + some stealth evasions.

**Comment by itsdarrylnorris** [comment]
I have created a simple PR to avoid this particular issue. https://github.com/berstend/puppeteer-extra/pull/240. If anyone can give that PR a quick look and test it out I will appreciated.

@brunogaspar Let me know if there is anything I can help getting your fork public, feel free to shoot me an email.

Hopefully, this https://github.com/berstend/puppeteer-extra/issues/238 gets resolve soon, so we can all contribute to the same project.

**Comment by berstend** [comment]
Hi all, as mentioned in #238 I'm working my way through the outstanding issues and PRs now.

Making `puppeteer-extra` compatible with the latest Puppeteer release is among the highest priorities. I'm being slowed down by the CI testing a bit, as we have quite an elaborate testing matrix currently to ensure backwards compatibility.

As [mentioned here](https://github.com/berstend/puppeteer-extra/pull/240#issuecomment-654325111) it could make sense to simplify this in the future by having the latest `puppeteer-extra` version always support the latest `puppeteer` version (and encouraging version pinning in case older versions are required).

TL;DR: I'm in the process of merging/testing PRs to support Puppeteer 5 asap. 😄

**Comment by scottweinert** [comment]
@berstend thanks for giving this some thought, and @itsdarrylnorris for the PR. Since this has been merged into master, I was looking to install the stealth package at this commit. However, since it uses Lerna's package system, do we need to wait for a release before we'd be able to properly install this as a dependency? Thanks

**Comment by berstend** [comment]
> @berstend thanks for giving this some thought, and @itsdarrylnorris for the PR. Since this has been merged into master, I was looking to install the stealth package at this commit. However, since it uses Lerna's package system, do we need to wait for a release before we'd be able to properly install this as a dependency? Thanks

I had to fix a Travis CI related issue first to ensure all tests go green for both puppeteer 5.0.0 and 2.1.1. (#243) 😅 

Will do some last checks and push a new version with Puppeteer 5.0.0 compatibility to npm.

**Comment by berstend** [comment]
It's now live on npm! 😄 🎉 

More specifically these are the new packages:

```
Successfully published:
 - puppeteer-extra-plugin-adblocker@2.11.4
 - puppeteer-extra-plugin-anonymize-ua@2.2.10
 - puppeteer-extra-plugin-block-resources@2.2.5
 - puppeteer-extra-plugin-click-and-wait@2.2.5
 - puppeteer-extra-plugin-devtools@2.2.10
 - puppeteer-extra-plugin-flash@2.2.6
 - puppeteer-extra-plugin-font-size@2.2.5
 - puppeteer-extra-plugin-recaptcha@3.1.12
 - puppeteer-extra-plugin-repl@2.2.5
 - puppeteer-extra-plugin-stealth@2.4.11
 - puppeteer-extra-plugin-user-data-dir@2.2.5
 - puppeteer-extra-plugin-user-preferences@2.2.5
 - puppeteer-extra-plugin@3.1.5
 - puppeteer-extra@3.1.11
lerna success published 14 packages
```

That means the following will now result in Puppeteer 5 + Extra & Stealth:

```
yarn add puppeteer puppeteer-extra puppeteer-extra-plugin-stealth
```


All [tests](https://travis-ci.org/github/berstend/puppeteer-extra/builds/706194343) a green for both  puppeteer `5.0.0` and `2.1.1`. Please report any issues you might find in this ticket.

Apologies again for the delay on this. Since today we have [6 awesome collaborators](https://github.com/berstend/puppeteer-extra/issues/238#issuecomment-655314232) onboard and we will revamp the CI so packages will be automatically published when the code is being updated (so I'm not the sole dependency on pushing to npm).

That means updates (stealth, playwright, etc) will come a lot faster.

Please head over to this [meta ticket](https://github.com/berstend/puppeteer-extra/issues/238) to stay up to date on the overall progress and feel free to share your ideas on what the important next steps might be.

**Comment by berstend** [comment]
Closing for now, please re-open in case of `Puppeteer 5` related issues.
